internal/fnruntime/runner.go

```diff
@@ -137,12 +137,18 @@ func (fr *FunctionRunner) do(input []*yaml.RNode) (output []*yaml.RNode, err err
 		return output, resultErr
 	}
 	if err != nil {
-		var fnErr *ExecError
-		if goerrors.As(err, &fnErr) {
-			fnResult.ExitCode = fnErr.ExitCode
-			fnResult.Stderr = fnErr.Stderr
-			fnErr.FnResult = fnResult
+		var execErr *ExecError
+		var stderrNotEmpty *StderrNotEmpty
+		switch {
+		case goerrors.As(err, &execErr):
+			fnResult.ExitCode = execErr.ExitCode
+			fnResult.Stderr = execErr.Stderr
 			fr.fnResults.ExitCode = 1
+		case goerrors.As(err, &stderrNotEmpty):
```

Comment:
I am feeling a bit uncomfortable retrieving this info through an error mechanism, but not able to think a way around it.

Reply:
Your new proposal should address this concern.

Reply:
done